src/Features/CSharp/Portable/BraceCompletion/AbstractCurlyBraceOrBracketCompletionService.cs

```diff
@@ -115,18 +114,22 @@ private static bool ContainsOnlyWhitespace(SourceText text, int openingPosition,
             if (closingPointLine - openingPointLine == 1)
             {
                 // Handling syntax tree directly to avoid parsing in potentially UI blocking code-path
-                var closingToken = document.Root.FindTokenOnLeftOfPosition(context.ClosingPoint);
-                var newLineString = options.FormattingOptions.NewLine;
-                newLineEdit = new TextChange(new TextSpan(closingToken.FullSpan.Start, 0), newLineString);
+                var closingToken = document.Root.FindToken(closingPoint - 1);
 
-                var generator = document.LanguageServices.GetRequiredService<SyntaxGeneratorInternal>();
-                var endOfLine = generator.EndOfLine(newLineString);
+                var newClosingToken = closingToken.WithPrependedLeadingTrivia(
+                    SpecializedCollections.SingletonEnumerable(SyntaxFactory.EndOfLine(options.FormattingOptions.NewLine)));
 
-                var rootToFormat = document.Root.ReplaceToken(closingToken, closingToken.WithPrependedLeadingTrivia(endOfLine));
-                document = document.WithChangedRoot(rootToFormat, cancellationToken);
+                var rootToFormat = document.Root.ReplaceToken(closingToken, newClosingToken);
+
+                newClosingToken = rootToFormat.FindToken(closingPoint - 1);
 
                 // Modify the closing point location to adjust for the newly inserted line.
-                closingPoint += newLineString.Length;
+                closingPoint = newClosingToken.Span.End;
+
+                var textChangeLength = newClosingToken.Span.End - closingToken.Span.End;
+                newLineEdit = new TextChange(new TextSpan(closingToken.FullSpan.Start, 0), newClosingToken.ToFullString()[..textChangeLength]);
```

Comment:
ok.  trying to reason about this.  we're saying that we go to the start of where the close token was.  then we figure out the distance the close token moved.  Then we grab that distance from the start of teh new closing token?

I'm not really getting it.  Can we be explicit about what text we're trying to grab here?  It looks like it's just supposed to be the potential newlines at the start of the close token?  Can we just grab this from it's trivia instead?  Or, even better, can we just grab from the source-text?

The reason i don't like this is that `..TextChangeLength` looks like it could grab random stuff (including the token contents itself).